File: driver.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <iterator> 
#include <vector>
#include "inventory.hpp"
#include "battle.hpp"
#include "wildEngimon.hpp"
#include "gameMap.hpp"
#include <set>
using namespace std;

/*Compile dengan*/
/*
g++ driver.cpp engimon.cpp skill.cpp wildEngimon.cpp gameMap.cpp tile.cpp

, player.cpp, inventory.cpp*/
int main()
{
	skill::InitializeSkillDatabase();
	engimon::IntializeDatabase();
	// string elem[5] = {"Fire", "Water", "Electric", "Ice"};
	// string lol[5] = {"Fire"};
	// skill a("a", 50, 0, lol, "0");
	// skill* s;
	// s = new skill[4];
	// skill nullSkill;
	// // s[0].printAll();
	// // s[1] = a;
	// // s[1].printAll();

	

	// // for (auto i = skill::skill_database.begin(); i != skill::skill_database.end(); ++i)
 // //        (*i).printAll();

	


	// engimon nullEngimon;
	// engimon Gab("Gab", &nullEngimon, &nullEngimon, "Magikarp", 4, 0);
	// engimon test;
	// test = Gab;
	// Gab.learnMove("tackle");
	// Gab.learnMove("bubble");
	// Gab.showAttributes();
	// cout << endl;
	// //test.showAttributes();
	// Gab.cry();

	// engimon Pulu("Pulu", &nullEngimon, &nullEngimon, "Magikarp", 10, 50);
	// engimon Pulu2("Pulu", &nullEngimon, &nullEngimon, "Vulpichu", 13, 20);
	// engimon Pulu3("Pulu", &nullEngimon, &nullEngimon, "Vulpichu", 2, 3);
	// Pulu.learnMove("bubble");
	// Pulu3.learnMove("tackle");
	// Pulu3.printMoves();

	// cout << endl;
	// wildEngimon WILDE("Pikachu",20);
	// WILDE.showAttributes();
	

	// player newPlayer("coba", Pulu, 50, 0, 1, 0, 0);

	// newPlayer.showEngimonDetails();
	// cout << endl <<Pulu3.getLevel();
	// Battle b2(&newPlayer.getActiveEngimon(), &Pulu3, &newPlayer);
	// b2.initiateBattle();
	//newPlayer.getActiveEngimon().getName();

	// Battle b1(&Pulu, &Pulu2, &newPlayer);
	// cout << "sini?" <<endl;
	// b1.initiateBattle();
	// cout << "masuk kah?" <<endl;
	// newPlayer.addEngimon(Pulu2);  

	// newPlayer.interact();       

	// newPlayer.deleteEngimonSelect();
	// newPlayer.showEngimonDetails();
	
	try{
	gameMap g("inputMapFile.txt");
	g.generateEngimon();
	g.updateMap(5,5,6,5);
	g.printMap();
}
	catch(char const* e)
	{
		cout << e;
	}

	// ifstream f("inputMapFile.txt");
	// int a, b;
	// char c;
	// f >> a;
	// f >> b;

	// cout << a << ", " << b << endl;
	// f = ifstream("inputMapFile.txt");
	// while(f >> c)
	// {
	// 	cout << c;
	// }
	return 0;
}

File: Java/Driver.java
public class Driver {
    public static void main(String[] args) {
        database_skill database_skill = new database_skill();
        database_engimon database_engimon = new database_engimon();
//        String[] elem = {"Fire", "Water", "Electric", "Ice"};
//        String[] lol = {"Fire"};
//        skill a = new skill("a", 50, 0, lol);
//        skill[] s;
//        s = new skill[4];
//        skill nullSkill = new skill();
//        //s[0].printAll();
//        s[1] = a;
//        s[1].printAll();
//
//        System.out.println("\n");
//        try
//        {
//         s[0] = database_skill.random("Fire","Electric");
//         s[0].printAll();
//        }
//
//        catch(Exception e)
//        {
//            System.out.println(e.getMessage());
//        }

        // System.out.println(database_skill.isExist("splash"));

        // System.out.println(database_skill.skillUniqueEngimon("Magikarp") == null);
//        System.out.println(database_engimon.isExist("Magikarp"));
//        System.out.println("");
//        player_engimon Gab;
//        try
//        {
//            Gab = new player_engimon("Gab", null, null, "Magikarp", 4, 0);
//            engimon test;
//            test = Gab;
//            Gab.showAttributes();
//            // System.out.println(Gab == null);
//            Gab.learnMove("tackle");
//            Gab.learnMove("bubble");
//            System.out.println();
//            test.showAttributes();
//            Gab.cry();
//
//            System.out.println("");
//            wild_engimon we = new wild_engimon("mountains");
//            we.showAttributes();;
//            we.cry();
//        }
//        catch (NullPointerException n)
//        {
//            n.printStackTrace();
//        }
//        catch (Exception e)
//        {
//            System.out.println(e.getMessage());
//            e.printStackTrace();
//        }

        gameMap g;
        try {
            g = new gameMap("inputMapFile.txt");
            g.updateMap(g.getMapLength()/2, g.getMapWidth()/2,
                    g.getMapLength()/2, g.getMapWidth()/2 + 1);
            g.printMap();
            map_visualizer m = new map_visualizer(g);

            System.out.println("x: " + g.tile_map.elementAt(9).elementAt(11).x);
            System.out.println("y: " + g.tile_map.elementAt(9).elementAt(11).y);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }

//        System.out.println('h');

//
//        try
//        {
//            g.generateEngimon();
//            g.updateMap(g.getMapLength()/2, g.getMapWidth()/2,
//                    g.getMapLength()/2, g.getMapWidth()/2 + 1);
//        }
//        catch (Exception e)
//        {
//            e.printStackTrace();
//            System.out.println(e.getMessage());
//        }
//
//        g.printMap();

    }
}
